lib/radial_pf.m

```diff
@@ -0,0 +1,94 @@
+function [mpc, success, iterations] = radial_pf(casedata,mpopt)
+%RADIAL_PF  Solves the power flow using a backward-forward sweep method.
+%
+%   [mpc, success, iterations] = radial_pf(casedata,mpopt)
+%
+%   Inputs:
+%       casedata : either a MATPOWER case struct or a string containing
+%           the name of the file with the case data
+%       mpopt : MATPOWER options struct to override default options
+%           can be used to specify the solution algorithm, output options
+%           termination tolerances, and more.
+%
+%   Outputs:
+%       mpc : results struct with all fields from the input MATPOWER case,
+%             with solved voltages, active and reactive power flows
+%             and generator active and reactive power output.
+%       success : success flag, 1 = succeeded, 0 = failed
+%       iterations : number of iterations
+%
+%  See also CASEFORMAT, LOADCASE, MPOPTION.
+
+%% load case data
+mpc = loadcase(casedata);
+nb = size(mpc.bus,1);
+nl = size(mpc.branch,1);
+if nb - nl ~= 1
```

Comment:
Good. I forgot to mention that I think it should be a fatal error (i.e. use `error()` instead of `fprintf()` and `success = 0`) when the network is not radial. Also, as a best practice, I try never to use `return` in the middle of a function.